### DISCUSSION TITLE
Strawberry Shake non-singleton usage

### DISCUSSION THREAD

**Discussion by adamhathcock** [ChilliCream/hotchocolate#3507] [Ideas]
There doesn't seem to be a way to override the settings to inject any kind of transient behavior.  Mainly, I want to use a different HTTP header on each request as I want the shake client to be a stateless client for a server.

It looks like the intended usage is a singleton in an application for caching and whatnot.  I recognize I shouldnt use any of this but there doesn't appear to be hook for editing the HTTP requests to add auth information.

I'm happy to try to add this myself with a PR but I don't think I know enough about the intentions here.

**Comment by adamhathcock** [comment]
Looks like this issue might help me https://github.com/ChilliCream/hotchocolate/issues/3467

**Comment by ecrocombe** [comment]
I too am in a similar boat.
I need to connect to the same service for subscriptions over 1000 times, 1 per customer.
I am in no control over the server implementation and is part of a separate organisation.
The server owners have asked me to connect once per customer for load balancing reasons.

A singleton client is not usable in this use case.
